File: editorial/chap08/python/answer_B51.py
# 입력
S = input()

# 왼쪽부터 순서대로 확인해 나간다
# 문자열은 0번째 문자부터 시작하는 것에 주의
Stack = []
for i in range(len(S)):
	if S[i] == '(':
		Stack.append(i + 1)
	if S[i] == ')':
		print(Stack.pop(), i + 1)

File: editorial/chap08/cpp/answer_B51.cpp
#include <iostream>
#include <stack>
using namespace std;
 
int main() {
	// 입력
	string S;
	cin >> S;
 
	// 왼쪽부터 순서대로 확인해 나간다
	// 문자열은 0번째 문자부터 시작하는 것에 주의
	stack<int> Stack;
	for (int i = 0; i < S.size(); i++) {
		if (S[i] == '(') {
			Stack.push(i + 1);
		}
		if (S[i] == ')') {
			cout << Stack.top() << " " << i + 1 << endl;
			Stack.pop();
		}
	}
	return 0;
}
